storey/table.py

```diff
@@ -74,3 +75,625 @@ async def persist_key(self, key, event_data_to_persist):
+                raise ValueError(f'Requested period for aggregate {aggr.name} does not match existing period at {self._table_path}. '
+                                 f"Requested: {aggr.windows.period_millis}, existing: {schema_aggr['period_millis']}")
+            requested_raw_aggregates = aggr.get_all_raw_aggregates()
+            existing_raw_aggregates = get_all_raw_aggregates(schema_aggr['aggregates'])
+            # validate if current feature store contains all aggregates needed for the requested calculations
+            if self._read_only and not requested_raw_aggregates.issubset(existing_raw_aggregates):
+                raise ValueError(
+                    f'Requested aggregates for feature {aggr.name} do not match with existing aggregates at {self._table_path}. '
+                    f"Requested: {aggr.aggregations}, existing: {schema_aggr['aggregates']}")
+            # Check if more raw aggregates are requested, in which case a schema update is required
+            if not self._read_only and requested_raw_aggregates != existing_raw_aggregates:
+                should_update = True
+
+        return should_update
+
+    async def _save_key(self, key):
+        await self._storage._save_key(self._container, self._table_path, key, self.get_aggregations_attrs(key))
+
+    def _aggregates_to_schema(self):
+        schema = {}
+        for aggr in self._aggregates:
+            if isinstance(aggr.windows, SlidingWindows):
+                window_type = "SlidingWindow"
+            else:
+                window_type = "FixedWindow"
+            schema[aggr.name] = {'period_millis': aggr.windows.period_millis,
+                                 'aggregates': list(get_all_raw_aggregates(aggr.aggregations)),
+                                 'window_type': window_type, 'max_window_millis': aggr.windows.max_window_millis}
+
+        return schema
+
+    def get_aggregations_attrs(self, key):
+        if key in self._attrs_cache:
+            return self._attrs_cache[key].aggregations
+        else:
+            return None
+
+    def set_aggregations_attrs(self, key, element):
+        if key in self._attrs_cache:
+            self._attrs_cache[key].aggregations = element
+        else:
+            self._attrs_cache[key] = _CacheElement({}, element)
+
+    def get_static_attrs(self, key):
+        if key in self._attrs_cache:
+            return self._attrs_cache[key].static_attrs
+        else:
+            return None
+
+    def set_static_attrs(self, key, value):
+        if key in self._attrs_cache:
+            self._attrs_cache[key].static_attrs = value
+        else:
+            self._attrs_cache[key] = _CacheElement(value, None)
+
+    def get_keys(self):
+        return self._attrs_cache.keys()
+
+
+class _CacheElement:
+    def __init__(self, static_attrs, aggregations):
+        self.static_attrs = static_attrs
+        self.aggregations = aggregations
+
+
+class AggregatedStoreElement:
```

Comment:
Why did you moved all of this here as well (from this line forward)? doesn't seem to belong here

Reply:
leaving it in aggregations.py created cyclic dependency 
also it's internal to store impl, it makes sense it will be in the same place